### DISCUSSION TITLE
How to only apply compact to timeline on mobile?

### DISCUSSION THREAD

**Discussion by garth** [saadeghi/daisyui#3309] [Q&A] [answered]
It seems that the timeline component is not following mobile first principles, ie the compact version should be the default and there should be a modifier class that can be applied to make the timeline place content on both sides of the line.

Tailwind css v4 will add support for `not:*` so that we can do `not:lg:timeline-compact`, but until then is there a solution to only apply `timeline-compact` on smaller screens?

**Comment by garth** [comment, marked as answer]
It seems that a workaround is to add a set of "desktop first" breakpoints to the tailwind config.

```ts
  theme: {
    extend: {
      screens: {
        'less-than-2xl': { max: '1535px' },
        'less-than-xl': { max: '1279px' },
        'less-than-lg': { max: '1023px' },
        'less-than-md': { max: '767px' },
        'less-than-sm': { max: '639px' },
      },
    },
  },
```

Then we can do the following:

```html
<ul class="timeline timeline-vertical less-than-md:timeline-compact">
  ...
</ul>
```

Not ideal but it works

**Reply by saadeghi** [reply]
There's already such thing in Tailwind by default. `max-*`  
Like `max-lg:timeline-compact`

**Reply by garth** [reply]
> There's already such thing in Tailwind by default. `max-*` Like `max-lg:timeline-compact`

As I understood it, `max-*` is for range queries, so it only works between values (eg `sm:max-md:*`) and so cannot be used for mobile.

**Reply by saadeghi** [reply]
- `md:` md and larger
- `max-md:` md and smaller
- `sm:max-md:` larger than sm and smaller than md

For example see the code example for compact timeline here:  
https://daisyui.com/components/timeline/#timeline-with-icon-snapped-to-the-start

It uses `max-md:timeline-compact` which means it's compact for md and smaller
